### ISSUE TITLE
consider-using-f-string is triggered in a functional context

### ISSUE THREAD

**Issue by sam-s** [PyCQA/pylint#8224]
### Bug description

Pylint suggest f-string in a functional context where it cannot be used:
```
map("{:,d}".format, [1,10,100,1000,10000])
```
The alternative:
```
(f"{x:,d}" for x in [...]) 
```
while, admittedly, shorter by a few characters, is not more clear.

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
[C0209(consider-using-f-string), ] Formatting a regular string which could be a f-string
```

### Expected behavior

no diagnostics

### Pylint version

```shell
pylint 2.15.10
astroid 2.13.5
Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]
```

### OS / Environment

ubuntu

**Comment by Pierre-Sassoulas** [comment]
This check is not about readability or terseness, but after a quick check, I'm surprised that the map version using format is actually faster:
```
python -m timeit '(f"{x:,d}" for x in [1,10,100,1000,10000])'
2000000 loops, best of 5: 167 nsec per loop
python -m timeit '(map("{:,d}".format, [1,10,100,1000,10000]))'
2000000 loops, best of 5: 119 nsec per loop
```

**Comment by clavedeluna** [comment]
I also consistently get that using map is faster than f-string equivalent. I'm not sure there's much more investigation we can do. Sounds like deciding if we want pylint to be consistent and always suggest using f-string or to prioritize map's performance and tell pylint to not emit `consider-using-fstring` in cases when `map` is used

**Comment by sam-s** [comment]
> `(f"{x:,d}" for x in [1,10,100,1000,10000])` is insignificantly shorter than `(map("{:,d}".format, [1,10,100,1000,10000]))` and subjectively easier to read.

How would you rewrite
```
pandas_series.map("{:.2%}".format)
```

**Comment by sam-s** [comment]
what about [`display.float_format`](https://pandas.pydata.org/docs/user_guide/options.html)?
```
with pd.option_context("display.float_format", '{:.4g}'.format):
    ...
```
